src/es6.js

```diff
@@ -7,50 +7,48 @@
 //----------------
 // const, =>, default parameters, arrow functions default return statements using ()
 
-var food = 'pineapple';
+const food = 'pineapple';
 
-var isMyFavoriteFood = function(food) {
-  food = food || 'thousand-year-old egg'; //This sets a default value if `food` is falsey
-  return food === 'thousand-year-old egg';
-};
+const isMyFavoriteFood = (food = 'thousand-year-old egg') => food;
 
-var isThisMyFavorite = isMyFavoriteFood(food);
+const isThisMyFavorite = isMyFavoriteFood(food);
 
 //----------------
 //const, class, template literals, enhanced object literals (foo: foo, -> foo,)
 
-var User = function(options) {
-  this.username = options.username;
-  this.password = options.password;
-  this.sayHi = function() {
-    return this.username + ' says hello!';
-  };
+class User {
+  constructor(options) {
+    this.username = options.username;
+    this.password = options.password;
+    this.sayHi = function () {
+      return this.username + ' says hello!';
```

Comment:
Don't attach the methods onto the object in the `constructor`, same thing that I mentioned above.  Also use a string template for the string being returned instead of using string concatenation.